lib/membrane_mp4/muxer/cmaf.ex

```diff
@@ -56,14 +56,14 @@ defmodule Membrane.MP4.Muxer.CMAF do
       key_frame? and state.old_input_caps != nil ->
         %{new_output_caps: new_output_caps, old_input_caps: old_input_caps} = state
 
-        {buffer, state} = generate_segment(old_input_caps, sample.metadata, state)
+        {buffer, state} = generate_segment(old_input_caps, sample, state)
         state = %{state | samples: [sample], old_input_caps: nil, new_output_caps: nil}
 
         {{:ok, buffer: {:output, buffer}, caps: {:output, new_output_caps}, redemand: :output},
          state}
 
-      key_frame? and sample.metadata.timestamp - state.elapsed_time >= state.segment_duration ->
-        {buffer, state} = generate_segment(caps, sample.metadata, state)
+      key_frame? and sample.dts - state.elapsed_time >= state.segment_duration ->
+        {buffer, state} = generate_segment(caps, sample, state)
```

Comment:
WDYT about passing here just `%{pts: sample.pts, metadata: sample.metadata}` instead of passing the whole sample and that performing some weird magic with an empty payload?

Reply:
It's kinda verbose, but I don't have any better ideas - we will go that way 😄